### PR TITLE
[wip] Initial devspace environment for running edgefarm

### PR DESCRIPTION
**Pull request by siredmar**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -32,7 +32,7 @@ There are some predefined handy commands that make the setup easier.
 
 `devspace run update`: Update all dependencies.
 
-`devspace run instanciate-nodes`: Instanciate some virtual kubeedge nodes.
+`devspace run instantiate-nodes`: Instantiate some virtual kubeedge nodes.
 
 `devspace run purge-nodes`: Destroys the virtual kubeedge nodes.
```
Comment (KairaMiu) [inline README.md:38]: How to access argo cd locally?

File: dev/applications/mixed-app-network/mixed-app-network.yaml [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+      type: edge-worker
+      properties:
+        image: nginx
+        name: simplenginx
+      traits:
+        - type: labels
+          properties:
+            "app": "edge-worker-2"
+    - name: network
+      type: application-network
```
Comment (KairaMiu) [inline dev/applications/mixed-app-network/mixed-app-network.yaml:35]: didn't i need to add usersnames here as well? How to enable users secrets here?

File: README.md (hunk 1 above)
Comment (KairaMiu) [inline README.md:38]: How to access keycloak?

File: devspace.yaml [hunk 3]
```diff
@@ -68,19 +68,102 @@ commands:
...
             - ${HOST_IP_NIP_IO}
+          websocket:
+            port: 10000
+          https:
+            port: 10002
+          stream:
+            port: 10003
+          tunnel:
+            port: 10004
+          annotations: {}
```
Comment (KairaMiu) [inline devspace.yaml:103]: why do you configure the values here? as this is already done in `charts/kubeedge/values.yaml`?

**Review by KairaMiu — CHANGES_REQUESTED** [review]
Gerneral:
Folder structure below `dev` shall also be in the form:
```
dev/helm/”put-your-dev-charts-here”
dev/manifests/”put-your-dev-manifests-here ('normal' and kustomize)"
```

File: dev/applications/mixed-app-network/mixed-app-network.yaml (hunk 2 above)
Comment (KairaMiu) [reply dev/applications/mixed-app-network/mixed-app-network.yaml:35]: please update network to full example:
```yaml
    - name: network
      type: application-network
      properties:
        usernames:
          - cloud-worker-1
          - edge-worker-1
          - edge-worker-2
```

Comment (KairaMiu) [reply dev/applications/mixed-app-network/mixed-app-network.yaml:35]: Better idea,... remove this example here and use `dev/manifests/applications/mixed-app-network/mixed-app-network.yaml` from [edgefarm.applications](https://github.com/edgefarm/edgefarm.applications) instead.

Comment (KairaMiu) [reply dev/applications/mixed-app-network/mixed-app-network.yaml:35]: or use some more of these examples?

Comment (siredmar) [reply dev/applications/mixed-app-network/mixed-app-network.yaml:35]: using dev/manifests/applications/mixed-app/mixed-app-network.yaml

File: devspace.yaml [hunk 4]
```diff
@@ -5,13 +5,13 @@ dependencies:
     source:
       git: https://github.com/edgefarm/devspace.base
       subPath: /environments/k3d
-      branch: main
+      branch: ingress-nginx
```
[the remark this answers is not in the document]
Comment (siredmar) [reply devspace.yaml:8]: Done with another PR when external PR is done

File: README.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (siredmar) [reply README.md:33]: done

Comment (siredmar) [reply README.md:33]: done

[the remark this answers is not in the document]
Comment (KairaMiu) [reply README.md:34]: awesome

Comment (KairaMiu) [reply README.md:34]: like it

Comment (KairaMiu) [reply README.md:34]: stil open, but for now ok.

**Pull request closed without merging**